pkg/tar/tar_test.go

```diff
@@ -531,3 +598,56 @@ func TestExtractTarStreamTimeout(t *testing.T) {
 		t.Errorf("Did not get the expected timeout error. err = %v", err)
 	}
 }
+
+func TestRoundTripTar(t *testing.T) {
```

Comment:
Test that we can round-trip creating and extracting TAR-ed content. This includes mixed-format archives with UTF-8 file headers.